internal/controller/util/cephfs_cg.go

```diff
@@ -307,3 +307,35 @@ func CheckImagesReadyToUse(
 
 	return true, nil
 }
+
+func GetVolumeSnapshotsOwnedByVolumeGroupSnapshot(
+	ctx context.Context,
+	k8sClient client.Client,
+	vgs *groupsnapv1beta1.VolumeGroupSnapshot,
+	logger logr.Logger,
+) ([]vsv1.VolumeSnapshot, error) {
+	volumeSnapshotList := &vsv1.VolumeSnapshotList{}
+	options := []client.ListOption{
+		client.InNamespace(vgs.Namespace),
+	}
+
+	if err := k8sClient.List(ctx, volumeSnapshotList, options...); err != nil {
+		return nil, err
+	}
+
+	logger.Info("GetVolumeSnapshotsOwnedByVolumeGroupSnapshot", "VolumeSnapshotList", volumeSnapshotList.Items)
+
+	var volumeSnapshots []vsv1.VolumeSnapshot
+
+	for _, snapshot := range volumeSnapshotList.Items {
+		for _, owner := range snapshot.ObjectMeta.OwnerReferences {
+			if owner.Kind == "VolumeGroupSnapshot" && owner.Name == vgs.Name {
```

Comment:
List is be a costly operation in larger cluster, we can extract the names like this https://github.com/ceph/ceph-csi/blob/72b9d5ac54d71e0f034a18a38d410a2da329c5e0/e2e/volumegroupsnapshot.go#L100-L101 or https://github.com/kubernetes-csi/external-snapshotter/blob/ba554e3965198005d755aa91855a60abe6492118/pkg/common-controller/groupsnapshot_controller_helper.go#L730

Reply:
@Madhu-1, thanks. we'll probably make the change in 4.19.  Just a note, we're only retrieving the list of snapshots within a specific namespace. Your suggestion relies on the name of the snapshot starts with `snapshot...`. That would mean, if the CSI changes the name pattern, then Ramen would fail.

Reply:
Yes that's correct, i agree it could be a breaking change at the same time if csi doesn't set the ownerRef and switch to another approach it will fail as well.  at the same time listing in a namespace can also become costly when we have more volumesnapshots in a namespace we could use ` client.MatchingField`? to match the ownerRef fields (not sure it works but just an idea)?